library/network_state.py

```diff
@@ -39,12 +39,22 @@ def __init__(self, module, module_name):
         self.params = module.params
         self.result = dict(changed=False)
         self.module_name = module_name
-        self.previous_state = libnmstate.show()
+        if hasattr(libnmstate, "show_running_config") and callable(
+            getattr(libnmstate, "show_running_config")
+        ):
+            self.previous_state = libnmstate.show_running_config()
+        else:
+            self.previous_state = libnmstate.show()
 
     def run(self):
         desired_state = self.params["desired_state"]
         libnmstate.apply(desired_state)
-        current_state = libnmstate.show()
+        if hasattr(libnmstate, "show_running_config") and callable(
+            getattr(libnmstate, "show_running_config")
+        ):
+            current_state = libnmstate.show_running_config()
+        else:
+            current_state = libnmstate.show()
```

Comment:
just a nit, as this is being used in two different places, what do you think about putting this block in its own function? That function should return what `show()` or `show_running_config()` is returning.

What do you think?

Reply:
I updated.